core/vm/evm.go

```diff
@@ -508,7 +508,7 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		return nil, common.Address{}, 0, vmerrs.ErrContractAddressCollision
 	}
 	// If the allow list is enabled, check that [evm.TxContext.Origin] has permission to deploy a contract.
-	if evm.chainRules.IsPrecompileEnabled(precompile.ContractDeployerAllowListAddress) {
+	if evm.chainRules.IsPrecompileEnabled(deployerallowlist.ContractAddress) {
```

Comment:
It seems like we are using the name `deployerallowlist.ContractAddress` here but `feemanager.ConfigKey` in `core/tx_pool.go`. Could we standardize this?

Reply:
hmm I think AWM stuff needs this to be a contract address for predicates (rather than the config keys). We should probably revert this to use contract addresses.

Reply:
reverted to use contract addresses